crates/precompiles/src/stablecoin_exchange/mod.rs

```diff
@@ -264,13 +264,29 @@ impl<'a, S: PrecompileStorageProvider> StablecoinExchange<'a, S> {
         Ok(())
     }
 
+    fn ensure_transfer_authorized(&mut self, token: Address, user: Address) -> Result<()> {
+        if token == PATH_USD_ADDRESS {
+            PathUSD::new(self.storage)
+                .token
+                .ensure_transfer_authorized(user, self.address)?;
+        } else {
+            TIP20Token::from_address(token, self.storage)?
+                .ensure_transfer_authorized(user, self.address)?;
+        }
```

Comment:
Now that PathUSD is treated like any other TIP20, we can just use `TIP20Token::from_address(token, self.storage)?.ensure_transfer_authorized(user, self.address)?;`. With this in mind, we probably can also remove the extra `ensure_transfer_authorized` function and jus tuse `TIP20.ensure_transfer_authorized` inline instead.